File: Sources/Gamified/ViewModels/StatsViewModel.swift
//
//  StatsViewModel.swift
//  
//
//  Created by Nathan Fallet on 27/01/2023.
//

import Foundation

class StatsViewModel: ObservableObject {
    
    // Configuration
    
    let service: GamifiedService
    let counters: [Counter]
    let isGridEnabled: Bool
    
    @Published var graphs: [Graph]
    
    // Initializer
    
    init(
        service: GamifiedService,
        counters: [Counter],
        isGridEnabled: Bool
    ) {
        self.service = service
        self.counters = counters
        self.isGridEnabled = isGridEnabled
        self.graphs = []
        DispatchQueue.global().async {
            let graphsData = service.getGraphs(startDate: .previous17Weeks, endDate: Date())
            DispatchQueue.main.async {
                self.graphs = graphsData
            }
        }
    }
    
}

File: Gamified/src/main/java/me/nathanfallet/gamified/viewmodels/StatsViewModel.kt
package me.nathanfallet.gamified.viewmodels

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.viewModelScope
import kotlinx.coroutines.launch
import me.nathanfallet.gamified.extensions.previous17Weeks
import me.nathanfallet.gamified.models.Counter
import me.nathanfallet.gamified.models.Graph
import me.nathanfallet.gamified.services.GamifiedService
import java.util.Date

class StatsViewModel(
    application: Application,
    val counters: List<Counter>,
    val isGridEnabled: Boolean
) : AndroidViewModel(application) {

    // Configuration

    private val _graphs = MutableLiveData<List<Graph>>()
    val graphs: LiveData<List<Graph>> = _graphs

    // Initializer

    init {
        viewModelScope.launch {
            _graphs.value = GamifiedService.shared.getGraphs(previous17Weeks, Date())
        }
    }


}